lte/cloud/go/services/lte/obsidian/models/conversion.go

```diff
@@ -144,6 +144,28 @@ func (m *NetworkRanConfigs) GetFromNetwork(network configurator.Network) interfa
 	return iCellularConfig.(*NetworkCellularConfigs).Ran
 }
 
+func (m *NetworkNgcConfigs) ToUpdateCriteria(network configurator.Network) (configurator.NetworkUpdateCriteria, error) {
+	iCellularConfig := orc8rModels.GetNetworkConfig(network, lte.CellularNetworkConfigType)
+	if iCellularConfig == nil {
+		return configurator.NetworkUpdateCriteria{}, fmt.Errorf("No cellular network config found")
+	}
+	iCellularConfig.(*NetworkCellularConfigs).Ngc = m
+	return orc8rModels.GetNetworkConfigUpdateCriteria(network.ID, lte.CellularNetworkConfigType, iCellularConfig), nil
+}
+
+func (m *NetworkNgcConfigs) GetFromNetwork(network configurator.Network) interface{} {
+	iCellularConfig := orc8rModels.GetNetworkConfig(network, lte.CellularNetworkConfigType)
+	if iCellularConfig == nil {
+		return nil
+	}
+	return iCellularConfig.(*NetworkCellularConfigs).Ngc
+}
+
+func (m *NetworkNgcConfigs) ConvertSuciEntsToProtos(ent *SuciProfile) *protos.SuciProfile {
+	suciData := &protos.SuciProfile{HomeNetworkPublicKeyIdentifier: ent.HomeNetworkPublicKeyIdentifier, HomeNetworkPublicKey: ent.HomeNetworkPublicKey, HomeNetworkPrivateKey: ent.HomeNetworkPrivateKey, ProtectionScheme: protos.SuciProfile_ECIESProtectionScheme(protos.SuciProfile_ECIESProtectionScheme_value[ent.ProtectionScheme])}
```

Comment:
Construct this literal with one line per field

Reply:
done